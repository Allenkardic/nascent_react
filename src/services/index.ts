// export const { useGetPokemonByNameQuery } = pokemonApi;
export { default as pokemonApi } from './pokemon';
