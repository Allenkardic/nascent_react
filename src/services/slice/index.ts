export * from './pokemonSlice';
export * from './singlePokemonSlice';
export * from './savedPokemonSlice';
