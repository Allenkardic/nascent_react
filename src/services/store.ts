import { configureStore, Action } from '@reduxjs/toolkit';
import { useDispatch, useSelector } from 'react-redux';
import { persistStore, persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import { ThunkAction } from 'redux-thunk';

import rootReducer, { RootState } from './root-reducer';

const persistConfig = {
  key: 'root',
  storage,
  whitelist: ['pokemon'],
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
  reducer: persistedReducer,
  middleware: getDefaultMiddleware =>
    getDefaultMiddleware({
      serializableCheck: false,
      immutableCheck: false,
    }),
});

export type AppDispatch = typeof store.dispatch;
export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector = () => useSelector;

export type AppThunk = ThunkAction<void, RootState, unknown, Action<string>>;
export const persistor = persistStore(store);
