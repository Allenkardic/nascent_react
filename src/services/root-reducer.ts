import { combineReducers } from '@reduxjs/toolkit';

import { pokemonSliceReducer, singlPokemonSliceReducer, savedPokemonSliceReducer } from './slice';

const rootReducer = combineReducers({
  pokemon: pokemonSliceReducer,
  singlePokemon: singlPokemonSliceReducer,
  savedPokemon: savedPokemonSliceReducer,
});

export type RootState = ReturnType<typeof rootReducer>;

export default rootReducer;
