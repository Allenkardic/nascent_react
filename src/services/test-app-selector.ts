import { pokemonSliceReducer, singlPokemonSliceReducer, savedPokemonSliceReducer } from './slice';
const state = {
  pokemon: pokemonSliceReducer,
  singlePokemon: singlPokemonSliceReducer,
  savedPokemon: savedPokemonSliceReducer,
};

export const testAppSelector = (f: any) => f(state);
