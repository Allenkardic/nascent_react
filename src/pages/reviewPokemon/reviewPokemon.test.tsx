import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import { ReviewPokemon } from '..';
import { store } from '../../services';
import 'jest-styled-components';

test('it renders ReviewPokeman page correctly', () => {
  const { container } = render(
    <Provider store={store}>
      <BrowserRouter>
        <ReviewPokemon />
      </BrowserRouter>
    </Provider>,
  );
  expect(container).toMatchSnapshot();
});
