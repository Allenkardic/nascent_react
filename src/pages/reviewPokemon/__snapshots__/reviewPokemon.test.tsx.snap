// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
.c7 {
  padding-top: 16px;
  background-color: #FFFFFF;
  display: flex;
  flex-direction: column;
  margin-left: auto;
  margin-right: auto;
  padding-bottom: 150px;
}

.c8 {
  padding: 0px 16px;
}

.c1 {
  font-weight: 400;
  color: #333333;
  font-size: 15px;
  text-align: left;
  font-family: Medium;
}

.c12 {
  font-weight: 400;
  color: #4F4F4F;
  font-size: 24px;
  text-align: left;
  font-family: Medium;
}

.c19 {
  font-weight: 700;
  color: #FFFFFF;
  font-size: 16px;
  text-align: center;
  font-family: Medium;
}

.c16 {
  font-weight: 400;
  color: #828282;
  font-size: 16px;
  text-align: left;
  font-family: Medium;
}

.c17 {
  font-weight: 400;
  color: #090E29;
  font-size: 16px;
  text-align: right;
  font-family: Medium;
}

.c6 {
  font-weight: 700;
  color: #090E29;
  font-size: 13px;
  text-align: center;
  font-family: Medium;
}

.c18 {
  display: flex;
  display: webkit-flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  color: #2A3362;
  background-color: #2A3362;
  border: 1px solid #2A3362;
  height: 56px;
  width: 100%;
  border-radius: 4px;
  transition: all 0.3s ease;
  cursor: pointer;
  opacity: 1;
  margin-bottom: 0px;
}

.c0 {
  display: flex;
  display: webkit-flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100px;
  padding: 0px 160px;
  border-bottom: 1px solid #E9EBF5;
  background-color: transparent;
}

.c2 {
  margin-top: 32px;
}

.c4 {
  width: 80px;
}

.c5 {
  height: 4px;
  border-radius: 8px;
  background-color: #2A3362;
  margin-bottom: 4px;
}

.c3 {
  display: flex;
  display: webkit-flex;
  flex-direction: row;
  align-items: center;
  align-items: webkit-align-center;
  justify-content: center;
  column-gap: 16px;
}

.c9 {
  display: flex;
  display: webkit-flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin-left: auto;
  margin-right: auto;
  border: 1px solid #EEEEEE;
  border-radius: 4px;
  padding: 24px;
}

.c10 {
  display: flex;
  width: 100%;
}

.c11 {
  display: flex;
  display: webkit-flex;
  height: 42px;
  align-items: center;
  justify-content: center;
  margin-left: auto;
  margin-right: auto;
  width: 60%;
  border: 1px solid #F2F2F2;
  border-radius: 30px;
  background-color: #FFFFFF;
}

.c14 {
  margin-top: 16px;
  margin-bottom: 24px;
  width: 100%;
}

.c15 {
  display: flex;
  display: webkit-flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  align-content: center;
  width: 100%;
  margin-bottom: 16px;
}

.c13 {
  height: 100px;
  width: 100px;
  border-radius: 50%;
  background-color: #EEEEEE;
  margin-top: 24px;
  margin-bottom: 8px;
}

@media only screen and (min-width: 768px) {
  .c8 {
    width: 34vw;
    min-width: 500px;
    margin-left: auto;
    margin-right: auto;
  }
}

@media (min-width: 48rem) {
  .c1 {
    font-size: 15px;
  }
}

@media (min-width: 48rem) {
  .c12 {
    font-size: 24px;
  }
}

@media (min-width: 48rem) {
  .c19 {
    font-size: 16px;
  }
}

@media (min-width: 48rem) {
  .c16 {
    font-size: 16px;
  }
}

@media (min-width: 48rem) {
  .c17 {
    font-size: 16px;
  }
}

@media (min-width: 48rem) {
  .c6 {
    font-size: 13px;
  }
}

@media (max-width: 768px) {
  .c0 {
    padding: 0px 16px;
    height: 74px;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      Review
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <div
            class="c5"
          />
          <div
            class="c6"
            color="#090E29"
            fw="700"
            ta="center"
          >
            User details
          </div>
        </div>
        <div
          class="c4"
        >
          <div
            class="c5"
          />
          <div
            class="c6"
            color="#090E29"
            fw="700"
            ta="center"
          >
            Pokemon
          </div>
        </div>
        <div
          class="c4"
        >
          <div
            class="c5"
          />
          <div
            class="c6"
            color="#090E29"
            fw="700"
            ta="center"
          >
            Review
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      class="c7"
    >
      <div
        class="c8"
      >
        <div>
          <div
            class="c9"
          >
            <div
              class="c10"
            >
              <div
                class="c11"
              >
                <div
                  class="c12"
                  color="#4F4F4F"
                >
                  Tue Nov 07 2023
                </div>
              </div>
            </div>
            <img
              class="c13"
            />
            <div
              class="c14"
            >
              <div
                class="c15"
              >
                <div
                  class="c16"
                  color="#828282"
                >
                  Pokemon
                </div>
                <div
                  class="c17"
                  color="#090E29"
                  ta="right"
                />
              </div>
              <div
                class="c15"
              >
                <div
                  class="c16"
                  color="#828282"
                >
                  First name
                </div>
                <div
                  class="c17"
                  color="#090E29"
                  ta="right"
                />
              </div>
              <div
                class="c15"
              >
                <div
                  class="c16"
                  color="#828282"
                >
                  Last name
                </div>
                <div
                  class="c17"
                  color="#090E29"
                  ta="right"
                />
              </div>
              <div
                class="c15"
              >
                <div
                  class="c16"
                  color="#828282"
                >
                  Phone number
                </div>
                <div
                  class="c17"
                  color="#090E29"
                  ta="right"
                />
              </div>
              <div
                class="c15"
              >
                <div
                  class="c16"
                  color="#828282"
                >
                  Address
                </div>
                <div
                  class="c17"
                  color="#090E29"
                  ta="right"
                />
              </div>
            </div>
            <button
              btnht="56px"
              class="c18"
              type="button"
            >
              <div
                class="c19"
                color="#FFFFFF"
                fw="700"
                ta="center"
              >
                Done
              </div>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
